src/jsTest/kotlin/it/krzeminski/githubactionstyping/AjvSchemaValidationTest.kt

```diff
@@ -98,3 +99,25 @@ private suspend fun String.shouldNotBeValid(): String {
     readFile(path.join(badDir, this), utf8) shouldNot beValid()
     return this
 }
+
+private inline fun <R> Dir.use(block: (Dir) -> R): R {
```

Comment:
How about using [`use` from Okio](https://github.com/square/okio/blob/d169b4387c229704710dda4b48e8767937a6dd33/okio/src/commonMain/kotlin/okio/Okio.kt#L53)? 

I know it would mean adding a dependency on a library just for this single function, but I'd generally also use Okio for reading from the file system.

Reply:
`okio.use` can only be used with classes implementing `okio.Closeable`,
just as `kotlin.use` can only be used with classes implementing `kotlin.AutoCloseable`.

But actually I've just seen I can still use the `kotlin.use` to make this code much leaner, see the update.

Reply:
`okio.use` can only be used with classes implementing `okio.Closeable`,
just as `kotlin.use` can only be used with classes implementing `kotlin.AutoCloseable`.

But actually I've just seen I can still use the `kotlin.use` to make this code much leaner, see the update.

Reply:
That's better, thanks!